Realm/RLMSyncSessionRefreshHandle.mm

```diff
@@ -53,20 +53,38 @@ - (instancetype)initWithFullURLPath:(NSString *)urlPath
 }
 
 - (void)invalidate {
-    [self.timer invalidate];
-    self.user = nil;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.timer invalidate];
+        self.user = nil;
+    });
 }
 
 - (void)scheduleRefreshTimer:(NSTimeInterval)fireTime {
```

Comment:
From the rest of this code `fireTime` appears to contain a time interval from 1970 representing the time at which the token will expire? It'd be a lot clearer to convert this to only expose this from `RLMTokenDataModel` as an `NSDate` so we don't have to reason here about how to interpret this value.